api/query/concrete_query.go

```diff
@@ -91,6 +94,10 @@ type Not struct {
 // substring match per test.
 func (TestNamePattern) Size() int { return 1 }
 
+// Size of TestNamePattern has a size of 1: servicing such a query requires a
+// substring match per subtest.
+func (SubtestNamePattern) Size() int { return averageNumberOfSubtests }
```

Comment:
Why averageNumberOfSubtests? Could you explain it in the comment

Reply:
Size represents the number of operations needed to perform the given query.
For a substring match on each an every subtest, the average size of the operation will be the average number of subtests per test.